src/odb/include/odb/db.h

```diff
@@ -7476,30 +7549,99 @@ class dbModInst : public dbObject
 
   static dbModInst* getModInst(dbBlock* block_, uint dbid_);
 
-  std::string getName() const;
-
   std::string getHierarchicalName() const;
+
+  bool findModITerm(const char* name, dbModITerm*& ret);
+
+  dbSet<dbModITerm> getModITerms();
+
   // User Code End dbModInst
 };
 
+class dbModITerm : public dbObject
+{
+ public:
+  dbModInst* getParent() const;
+
+  void setNet(dbModNet* net);
+
+  dbModNet* getNet() const;
+
+  void setNextNetModiterm(dbModITerm* next_net_moditerm);
+
+  dbModITerm* getNextNetModiterm() const;
+
+  void setPrevNetModiterm(dbModITerm* prev_net_moditerm);
+
+  dbModITerm* getPrevNetModiterm() const;
+
+  void setNextEntry(dbModITerm* next_entry);
+
+  dbModITerm* getNextEntry() const;
+
+  // User Code Begin dbModITerm
+  void setSigType(dbSigType& type);
+  dbSigType getSigType();
+  void setIoType(dbIoType& type);
+  dbIoType getIoType();
```

Comment:
Why are these settable on the iterm - they should come from the corresponding lower level bterm